src/core/typechecker.ml

```diff
@@ -174,13 +174,27 @@ and infer_computation state = function
         ((ty1, ty1') :: (ty2, ty2') :: eqs') @ eqs
       in
       (ty2, List.fold_left fold eqs cases)
-  | Ast.Promise (op, abs, p, comp) ->
+  | Ast.Promise (None, op, abs, p, comp) ->
```

Comment:
It would be good to merge the two cases.

Reply:
Done